### PR TITLE
NonkTube Add new extractor

### PR DESCRIPTION
**Pull request by sarbajitsaha**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Added extractor for videos from https://www.nonktube.com
Example of video pages supported - https://www.nonktube.com/video/117836/asian-cute-girl-escaped-from-school-with-her-new-bf

Flake8 compliant 

In reference to issue https://github.com/rg3/youtube-dl/issues/8647

### CODE REVIEW

**Comment by dstftw** [comment]
Bother to read
>[adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections

and fix all issues.

**Comment by sarbajitsaha** [comment]
I was wondering if anything else is needed. I went through those two URLs.

I have added two tests, the code is Flake8 compliant and seems to work with both Python2 and 3.

I am returning id, title(two options to extract this), uploader(None if unable to extract otherwise the link to the user profile), thumbnail link, video URL(only format is mp4) and age limit is 18 as its an adult site. Do I need to add anything more?

File: youtube_dl/extractor/nonktube.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            'id': '118698',
+            'ext': 'mp4',
+            'title': '2 Japanese Girls Sharing One Dick Uncesnored',
+            'thumbnail': 'https://www.nonktube.com/media/videos/tmb_2/118698/default.jpg',
+            'uploader': 'https://www.nonktube.com/user/cumdrinkingwife',
+            'age_limit': 18
+        }
+    }]
+    DOWNLOAD_URL_TEMPLATE = 'https://cdn.nonktube.com/%s.mp4'
+    UPLOADER_URL_TEMPLATE = 'https://www.nonktube.com/user/%s'
```
Comment (dstftw) [inline youtube_dl/extractor/nonktube.py]: No need in these variables.

File: youtube_dl/extractor/nonktube.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        webpage = self._download_webpage(url, video_id)
+
+        video_url = self.DOWNLOAD_URL_TEMPLATE % str(video_id)
```
Comment (dstftw) [inline youtube_dl/extractor/nonktube.py]: `video_id` is already a string.

File: youtube_dl/extractor/nonktube.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+        title = self._html_search_regex(
+            [r'<title>(?P<title>.+?) - NonkTube.com</title>',
+             r'<h1 class=.*>(?P<title>.+?)</h1>'], webpage, 'title', group='title')
```
Comment (dstftw) [inline youtube_dl/extractor/nonktube.py]: No need in explicit group name.

File: youtube_dl/extractor/nonktube.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        'url': 'https://www.nonktube.com/video/118698/2-japanese-girls-sharing-one-dick-uncesnored',
+        'md5': 'bb4bdd0a61f591e2f5f7429dc860327e',
+        'info_dict': {
+            'id': '118698',
+            'ext': 'mp4',
+            'title': '2 Japanese Girls Sharing One Dick Uncesnored',
+            'thumbnail': 'https://www.nonktube.com/media/videos/tmb_2/118698/default.jpg',
+            'uploader': 'https://www.nonktube.com/user/cumdrinkingwife',
+            'age_limit': 18
+        }
```
Comment (dstftw) [inline youtube_dl/extractor/nonktube.py]: Remove all duplicated tests.

File: youtube_dl/extractor/nonktube.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        thumbnail = self._html_search_regex(
+            [r'<img src="(?P<thumbnail>.+?)" title="' + re.escape(title) + r'" alt="' + re.escape(title) + r'" width="[0-9]*" height="[0-9]*" />'],
+            webpage, 'thumbnail', fatal=False, group='thumbnail')
+
+        # only one link of format /user/ is present in every video which gives the uploader username
+        uploader = self._html_search_regex(
+            [r'<a href="/user/(?P<uploader_id>[0-9A-Za-z]+?)">'],
+            webpage, 'uploader', fatal=False, group='uploader_id')
+        if uploader is not None:
+            uploader = self.UPLOADER_URL_TEMPLATE % uploader
```
Comment (dstftw) [inline youtube_dl/extractor/nonktube.py]: `uploader_url`.

File: youtube_dl/extractor/nonktube.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        'info_dict': {
+            'id': '118698',
+            'ext': 'mp4',
+            'title': '2 Japanese Girls Sharing One Dick Uncesnored',
+            'thumbnail': 'https://www.nonktube.com/media/videos/tmb_2/118698/default.jpg',
+            'uploader': 'https://www.nonktube.com/user/cumdrinkingwife',
+            'age_limit': 18
+        }
+    }]
+    DOWNLOAD_URL_TEMPLATE = 'https://cdn.nonktube.com/%s.mp4'
```
Comment (dstftw) [inline youtube_dl/extractor/nonktube.py]: No hardcodes. Extraction should be implemented in terms of `NuevoBaseIE` instead.

Comment (sarbajitsaha) [inline youtube_dl/extractor/nonktube.py]: Does this class return a list of formats with urls in which the video can be downloaded? It seems that for this website, all the videos are in mp4 format and I can't even control the video resolution, so every video will be of that URL format only. So can I use this in that case?

```python
 video_url = 'https://cdn.nonktube.com/%s.mp4' % video_id
```

Comment (dstftw) [inline youtube_dl/extractor/nonktube.py]: You should not touch extraction at all. It's already implemented in `NuevoBaseIE`.

Comment (sarbajitsaha) [inline youtube_dl/extractor/nonktube.py]: Ok. I checked out the extractor for lovehomeporn which seems to be similar

```python
 info = self._extract_nuevo(
            'http://lovehomeporn.com/media/nuevo/config.php?key=%s' % video_id,
            video_id)
```

I found a similar url in the website - https://www.nonktube.com/media/nuevo/config.php?key=118636. However, unlike the former which showed the xml when I loaded the link in the browser, here it shows **"restricted area"** I tried passing the original video URL to extract_nuevo function which also didn't work. In both cases, it raised error.

Comment (dstftw) [inline youtube_dl/extractor/nonktube.py]: See how browser calls it.

Comment (sarbajitsaha) [inline youtube_dl/extractor/nonktube.py]: As soon as I press the play button, in the network tab(Chrome), I get this request 

Request URL:https://cdn.nonktube.com/117475.mp4?start=0

which seems to be the original video file link. That's why I structured it that way earlier. Can't seem to get any other format than mp4, nor can I change the resolution. I

**Pull request closed without merging**